libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/NavigationMapboxMap.java

```diff
@@ -0,0 +1,217 @@
+    this.mapboxMap = mapboxMap;
+    initializeRoute(mapView, mapboxMap);
+    initializeCamera(mapboxMap, mapboxNavigation);
+    initializeLocationLayer(mapView, mapboxMap);
+    initializeWayname(mapView, mapboxMap);
+  }
+
+  public void updateDefaultMapTopPadding(int topPadding) {
+    mapWayname.updateDefaultMapTopPadding(topPadding);
+  }
+
+  public void drawRoute(DirectionsRoute directionsRoute) {
+    mapRoute.addRoute(directionsRoute);
+  }
+
+  public void addMarker(Context context, Point position) {
+    Marker marker = createMarkerFromIcon(context, position);
+    mapMarkers.add(marker);
+  }
+
+  public void clearMarkers() {
+    removeAllMarkers();
+  }
+
+  public void updateLocation(Location location) {
+    locationLayer.forceLocationUpdate(location);
+    updateMapWaynameWithLocation(location);
+  }
+
+  public void updateCameraTrackingEnabled(boolean isEnabled) {
+    mapCamera.updateCameraTrackingLocation(isEnabled);
+  }
+
+  public void startCamera(DirectionsRoute directionsRoute) {
+    mapCamera.start(directionsRoute);
+  }
+
+  public void resumeCamera(Location location) {
+    mapCamera.resume(location);
+  }
+
+  public void resetCameraPosition() {
+    mapCamera.resetCameraPosition();
+  }
+
+  public void updateWaynameView(String wayname) {
+    mapWayname.updateWaynameLayer(wayname);
+  }
+
+  public void updateWaynameVisibility(boolean isVisible) {
+    mapWayname.updateWaynameVisibility(isVisible);
+  }
+
+  public void updateWaynameQueryMap(boolean isEnabled) {
+    mapWayname.updateWaynameQueryMap(isEnabled);
+  }
+
+  @SuppressLint("MissingPermission")
+  public void onStart() {
+    locationLayer.onStart();
+  }
+
+  public void onStop() {
+    locationLayer.onStop();
+  }
+
+  public void onDestroy() {
+    mapCamera.onDestroy();
+  }
+
+  private void initializeRoute(MapView mapView, MapboxMap map) {
+    Context context = mapView.getContext();
+    int routeStyleRes = ThemeSwitcher.retrieveNavigationViewStyle(context, R.attr.navigationViewRouteStyle);
+    mapRoute = new NavigationMapRoute(null, mapView, map, routeStyleRes);
+  }
+
+  private void initializeLocationLayer(MapView mapView, MapboxMap map) {
+    Context context = mapView.getContext();
+    int locationLayerStyleRes = ThemeSwitcher.retrieveNavigationViewStyle(context,
+      R.attr.navigationViewLocationLayerStyle);
+    locationLayer = new LocationLayerPlugin(mapView, map, null, locationLayerStyleRes);
+    locationLayer.setRenderMode(RenderMode.GPS);
+  }
+
+  private void initializeCamera(MapboxMap map, MapboxNavigation mapboxNavigation) {
+    mapCamera = new NavigationCamera(map, mapboxNavigation);
+  }
+
+  private void initializeWayname(MapView mapView, MapboxMap mapboxMap) {
+    initializeStreetsSource(mapboxMap);
+    MapWaynameLayoutProvider layoutProvider = new MapWaynameLayoutProvider(mapView.getContext());
+    LayerInteractor layerInteractor = new MapLayerInteractor(mapboxMap);
```

Comment:
@devotaaabel yeah for the tests like we discussed in the sprint 😅 they basically wrap different pieces of functionality from the `MapboxMap` - if I recall correctly, you pointed out that this would be a good idea to separate concerns.  Do you see a better way or one that involves less abstraction?

Reply:
Yes, remove them all and use the concrete classes. As commented before they're not adding any value 👀 
> Why is this `interface` needed? It's only implemented by `MapFeatureInteractor`. I think we can get rid of it. IMO it's not adding any value (we can still mock `MapFeatureInteractor` in tests). 👀 [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)